lib/mail/matchers/has_sent_mail.rb

```diff
@@ -51,6 +51,10 @@ def bcc(recipient_or_list)
         self
       end
 
+      def with_attachments(*attachments)
+        @attachments = attachments
+        self
+      end
```

Comment:
1. This API naming suggests that you should pass an array of attachments as the argument, which is how the other methods behave.
2. Chaining multiple `with_attachments` calls will overwrite the previous calls rather than accumulating them.

Consider something like
```ruby
def with_attachments(attachment_or_list)
  @attachments ||= []
  @attachments.concat Array(attachment_or_list)
end
```

+test for chained `with_attachments` calls.